src/veh_interact.cpp

```diff
@@ -1224,6 +1227,50 @@ bool veh_interact::overview( std::function<bool(const vehicle_part &pt)> enable,
     return redraw;
 }
 
+
+vehicle_part *veh_interact::get_most_damaged_part() const
+{
+    vehicle_part *high_part = nullptr;
+    int high_damage = 0;
+
+    for( size_t i = 0; i < veh->parts.size(); i++ ) {
+        auto *part = &veh->parts[i];
+        if( part->removed ) {
+            continue;
+        }
+
+        int dmg = part->base.damage();
+
+        if( dmg > high_damage ) {
+            high_part = part;
```

Comment:
This won't work very well: `high_damage` is never updated.
Actually, you can get rid of both `dmg` and `high_damage`. I'd also rename `high_part` to `result` or `res`:
```c++
if( res == nullptr || res.damage() < elem.damage() ) {
    res = &elem;
}
```